purchase_request/models/stock_move.py

```diff
@@ -67,7 +67,9 @@ def _action_cancel(self):
     def _compute_purchase_request_ids(self):
         for rec in self:
             rec.purchase_request_ids = rec.\
-                purchase_request_allocation_ids.mapped('purchase_request_id')
+            rec.purchase_request_ids = \
+                rec.purchase_request_allocation_ids.\
+                purchase_request_line_id.mapped('request_id')
```

Comment:
Hi @CasVissers 
You can get Singletone error
Try to use:
rec.purchase_request_allocation_ids.mapped(
                'purchase_request_line_id.request_id')